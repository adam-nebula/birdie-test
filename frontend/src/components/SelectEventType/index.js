import React, {Component} from "react";
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const useStyles = makeStyles((theme) => ({
    formControl: {
      margin: theme.spacing(1),
      minWidth: 300,
    },
  }));

export default function SelectEventType({eventType, handleEventTypeChange, eventTypes}){

    const classes = useStyles();

    return(
        <FormControl className={classes.formControl}>
            <InputLabel>Select an event</InputLabel>
            <Select
                value={eventType}
                onChange={handleEventTypeChange}>
                    {eventTypes.map(eventType => (
                        <MenuItem key={eventType.value} value={eventType.value}>
                            {eventType.label}
                        </MenuItem>
                    ))}
            </Select>
        </FormControl>
    )
}