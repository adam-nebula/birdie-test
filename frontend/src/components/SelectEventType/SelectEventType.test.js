import React from "react";
import renderer from 'react-test-renderer';
import SelectEventType from ".";

it('select event type renders corrrectly', () => {
    const tree = renderer.create(<SelectEventType 
        eventType="test_event_1"
        eventTypes={[{value: "test_event_1", label: "test_event_1"}, {value: "test_event_2", label: "test_event_2"}]}
        />).toJSON();
    expect(tree).toMatchSnapshot()
})