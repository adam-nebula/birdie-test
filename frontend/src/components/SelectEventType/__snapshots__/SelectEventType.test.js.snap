// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select event type renders corrrectly 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
  >
    Select an event
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      test_event_1
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="test_event_1"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
